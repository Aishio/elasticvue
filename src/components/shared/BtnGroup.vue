<template>
  <div :class="classes">
    <slot></slot>
  </div>
</template>

<script>
  export default {
    name: 'BtnGroup',
    props: {
      small: Boolean
    },
    computed: {
      classes () {
        return [
          'elevation-1',
          'btn-group',
          {'btn-group--small': this.small}
        ]
      }
    }
  }
</script>
