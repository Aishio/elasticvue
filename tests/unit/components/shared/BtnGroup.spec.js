import BtnGroup from '@/components/shared/BtnGroup'
import { mount } from '@vue/test-utils'

describe('components/shared/BtnGroup.vue', () => {
  it('should render correct default contents', () => {
    const wrapper = mount(BtnGroup)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should add the small group', () => {
    const wrapper = mount(BtnGroup, {
      propsData: {
        small: true
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
