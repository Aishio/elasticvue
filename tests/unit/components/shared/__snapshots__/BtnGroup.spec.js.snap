// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/shared/BtnGroup.vue should add the small group 1`] = `
<div
  class="elevation-1 btn-group btn-group--small"
/>
`;

exports[`components/shared/BtnGroup.vue should render correct default contents 1`] = `
<div
  class="elevation-1 btn-group"
/>
`;
